rotkehlchen/ethchain.py

```diff
@@ -53,17 +53,43 @@ def attempt_connect(self, ethrpc_port, mainnet_check=True) -> Tuple[bool, str]:
                         'the ethereum mainnet'.format(ethrpc_port)
                     )
                     return False, message
+                if self.web3.eth.syncing:  # pylint: disable=no-member
```

Comment:
unfortunately the `self.web3.eth.syncing` won't work reliably 100%.

The reason is not our fault but it's the way that the RPC call to the ethereum client method `eth_Syncing` responds. It's rather unreliable. For example for `geth` that is fast syncing if at some point it finds no peers `eth.syncing` will be `false` even if syncing has not finished.

For this PR this is going to be enough of a check but perhaps later either you or I can improve it by checking etherscan's latest block instead.

We do exactly that in raiden too. It starts [here](https://github.com/raiden-network/raiden/blob/e38a5f7ae5dda464663f4c2ea432d998deca204c/raiden/ui/cli.py#L150) and the relevant functions are [here](https://github.com/raiden-network/raiden/blob/e38a5f7ae5dda464663f4c2ea432d998deca204c/raiden/ui/cli.py#L182-L230)

**EDIT**: After looking at the code further below, I see you are doing something similar but with https://www.blockcypher.com/dev/ethereum/. Very nice! I did not even know about them. Are they as stable as etherscan?

Reply:
I'm not sure about the stability of blockcypher. I found the solution in [this StackExchange article](https://ethereum.stackexchange.com/questions/49726/how-do-i-get-the-latest-block-using-etherscan-api). It was simpler to use blockcypher so I used it.

Reply:
It's good enough I think. If it turns out to be unstable we can switch it out later. Thanks.